_posts/2022-11-23-#26152.md

```diff
@@ -7,17 +7,152 @@ authors: [Xekyo]
 components: ["wallet"]
 host: glozow
 status: upcoming
-commit:
+commit: 898ad9d590
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- The wallet *funds* a transaction by selecting inputs to cover its payment amount(s) and the fee
+  at the user's target feerate. This process is known as **coin selection**, which we have discussed in
+previous review clubs [#22009](/22009), [#17526](/17526) and [#17331](/17331).
+
+    - Notably, each candidate coin is considered using an "effective value," introduced in
+      [PR #17331](https://github.com/bitcoin/bitcoin/pull/17331). This deducts the cost to spend
+      this input at the target feerate from its `nValue`.
+
+- Since [PR #7600](https://github.com/bitcoin/bitcoin/pull/7600), the
+  [`BlockAssembler`](https://github.com/bitcoin/bitcoin/blob/aeb395dcdbfe2b1a6c77ff218939a18afde3add9/src/node/miner.h#L164)
+algorithm has used ancestor feerate, rather than individual feerate, to select transactions for
+inclusion in blocks. This strategy also enables users to fee-bump transactions using [Child Pays for
+Parent (CPFP)](https://bitcoinops.org/en/topics/cpfp/). We have discussed the `BlockAssembler`
+implementation in a [previous review club meeting](/24538).
+
+- On the flip side of CPFP, when creating a transaction at a target feerate, if the wallet selects
+  UTXOs from lower-feerate unconfirmed transactions, it may underestimate the amount of fees to put
+on this transaction. This issue has been an open problem for years; see
```

Comment:
I've expanded it a bit, there's a lot to cover here.
```suggestion
- The effective feerate of a transaction depends on the confirmation status of all of its parent transactions. If all parents are already confirmed, the effective feerate equals the feerate of the transaction itself. However, each unconfirmed parent transaction increases/decreases the effective feerate if the parent feerate was higher/lower than the target feerate. This behaviour is exactly the same as what we require for CPFP, but with CPFP we focus on boosting the parent transaction - whereas this PR deals with the wallet ensuring the child transaction receives the effective feerate the user intends. This issue has been an open problem for years; see
```

Reply:
I took a few issues with the wording but otherwise taken:

- the effective feerate depends on ancestors and all descendants of those ancestors, and not just parents
- a parent cannot increase the effective feerate of the transaction

Reply:
> the effective feerate depends on ancestors and all descendants of those ancestors, and not just parents

Whoops, yes I meant ancestors instead of parents - sorry for the confusion.

> a parent cannot increase the effective feerate of the transaction

Interesting, I hadn't considered that asymmetry (that it can only decrease) - but you're absolutely right.